### ISSUE TITLE
Running Release Manager tasks within a container image

### ISSUE THREAD

**Issue by justaugustus** [kubernetes/release#960]
@hoegaarden [mentioned](https://github.com/kubernetes/release/issues/958#issuecomment-563239639) that he uses a container image to run the Release Manager tasks.

[Cutting v1.15.0-alpha.2](https://docs.google.com/document/d/1Xv5w_eNvLvD-nNinMNqQAh0qlzee8btqAyHyFFMz3z4/edit?usp=sharing) doc mentioned in the Branch Manager handbook also references using as a container image for tasks.

It would be great to have one of these Dockerfiles (or an amalgam of the two) put up in the repo for other Release Managers to consider using.

/assign @hoegaarden
/sig release
/area release-eng
/milestone v1.18
/priority important-soon
/kind feature

**Comment by hoegaarden** [comment]
TL;DR: I cannot share all the things, so I don't think it is useful to PR that in.


I can share parts of the Dockerfile I am using, but not everything.
I have more stages that bring in more and it is pretty much tailored to my needs and circumstances and not meant to be generally shared and also has some stuff (more or less sensitive data) in it. That could probably all be parametrised but right now it is not.

So I don't think it is useful to PR that into the repo, as it is not generally useful / cannot be used as is. However it might be an inspiration for others, that's why parts of it can be found further down.

Right now it should also be possible to just use the `k8s-cloud-builder` (which holds all the essential tools) to base your image on and just add the your MTA, manage how to load your keys, and whatever you need to do.

Also, the linked doc holds pretty much what I have, which I believe is the thing I shared with @Bubblemelon back then.

/close
Please just reopen if you still think I should PR things in.

<details>
  <summary>(some parts of) the Dockerfile</summary>

```Dockerfile
##------------------------------------------------------------
FROM debian:sid as generic

ARG GO_VERSION='1.13.x'
ARG DEBIAN_FRONTEND=noninteractive
ARG GIT_AUTHOR_NAME
ARG GIT_AUTHOR_EMAIL
ARG RELEASE_SRC_DIR='/release'
ARG REPO_URL='https://github.com/kubernetes/release'

RUN apt-get -y update \
  && apt-get -y install gettext-base less git procps vim curl make lastpass-cli lsb-release jq bsdmainutils gnupg2 bc rsync python-pip \
  && mkdir -p ~/bin \
  && curl -sL -o ~/bin/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme \
  && chmod 0750 ~/bin/gimme \
  && echo 'running gimme' \
  && ~/bin/gimme "${GO_VERSION:-1.10.x}" >> "${HOME}/.bashrc" \
  && export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" \
  && echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list \
  && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - \
  && apt-get update -y && apt-get install google-cloud-sdk -y \
  && pip install --no-cache yq \
  && rm -rf /var/lib/apt/lists/*

ENV GIT_AUTHOR_NAME="${GIT_AUTHOR_NAME:-Jane Doe}"
ENV GIT_AUTHOR_EMAIL="${GIT_AUTHOR_EMAIL:-jane.doe@cool-employer.com}"

RUN git config --global user.name "$GIT_AUTHOR_NAME" \
  && git config --global user.email "$GIT_AUTHOR_EMAIL"
RUN mkdir -p "$RELEASE_SRC_DIR" \
  && git clone "$REPO_URL" "$RELEASE_SRC_DIR"

WORKDIR "$RELEASE_SRC_DIR"

ENTRYPOINT [ "bash" ]

##------------------------------------------------------------
FROM generic AS add-mta

ARG SMTP_SMART_HOST='fill me' # e.g. smtp.internal.cool-employer.com
ARG MAIL_DOMAIN='fill me' # e.g. cool-employer.com

RUN apt-get -y update \
  && apt-get -y install ssmtp \
  && rm -rf /var/lib/apt/lists/*

RUN chfn -f "${GIT_AUTHOR_NAME}" \
  && echo "root:${GIT_AUTHOR_EMAIL}:${SMTP_SMART_HOST}" > /etc/ssmtp/revaliases \
  && { \
       echo "root=postmaster" \
    && echo "mailhub=${SMTP_SMART_HOST}" \
    && echo "rewriteDomain=${MAIL_DOMAIN}" \
    && echo "FromLineOverride=YES" \
    && echo "UseTLS=Yes" \
    && echo "UseSTARTTLS=Yes" ; \
  } > /etc/ssmtp/ssmtp.conf

##------------------------------------------------------------
FROM add-mta
# ... more stages and stuff ...
```
</details>
